6_survival_analysis/nbconverted/expression_eda.py

```diff
@@ -0,0 +1,372 @@
+# 1 = observed non-silent mutation in this gene for this sample, 0 otherwise
+mutation_df = pancancer_data[1]
+print(mutation_df.shape)
+mutation_df.iloc[:5, :5]
+
+
+# ### Load copy number info
+# 
+# Get copy loss/gain info directly from GISTIC "thresholded" output. This should be the same as (or very similar to) what the Park et al. study uses.
+
+# In[7]:
+
+
+sample_freeze_df = pancancer_data[0]
+copy_samples = set(sample_freeze_df.SAMPLE_BARCODE)
+print(len(copy_samples))
+
+
+# In[8]:
+
+
+copy_loss_df, copy_gain_df = load_cnv_data(
+    cfg.data_dir / 'pancan_GISTIC_threshold.tsv',
+    copy_samples
+)
+print(copy_loss_df.shape)
+copy_loss_df.iloc[:5, :5]
+
+
+# In[9]:
+
+
+print(copy_gain_df.shape)
+copy_gain_df.iloc[:5, :5]
+
+
+# In[10]:
+
+
+sample_freeze_df.head()
+
+
+# ### Load expression data
+# 
+# We'll also subset to the top features by mean absolute deviation, if that option .
+
+# In[11]:
+
+
+if data_type == 'gene expression':
+    exp_df = pd.read_csv(gene_expression_data_file, sep='\t', index_col=0)
+elif data_type == 'rppa':
+    exp_df = pd.read_csv(rppa_data_file, sep='\t', index_col=0)
+    
+print(exp_df.shape)
+exp_df.iloc[:5, :5]
+
+
+# In[12]:
+
+
+# standardize features first
+exp_df = pd.DataFrame(
+    StandardScaler().fit_transform(exp_df),
+    index=exp_df.index.copy(),
+    columns=exp_df.columns.copy()
+)
+print(exp_df.shape)
+exp_df.iloc[:5, :5]
+
+
+# In[13]:
+
+
+# subset to subset_feats features by mean absolute deviation
+if subset_feats is not None:
+    mad_ranking = (
+        exp_df.mad(axis=0)
+               .sort_values(ascending=False)
+    )
+    top_feats = mad_ranking[:subset_feats].index.astype(str).values
+    print(top_feats[:5])
+    exp_mad_df = exp_df.reindex(top_feats, axis='columns')
+else:
+    exp_mad_df = exp_df
+    
+print(exp_mad_df.shape)
+exp_mad_df.iloc[:5, :5]
+
+
+# ### Get sample info and hit groups for gene/cancer type
+
+# In[14]:
+
+
+def get_hits_for_gene_and_tissue(identifier, cancer_classification):
+    """Given a gene and tissue, load the relevant mutation/CNV information,
+    and divide the samples into groups to compare survival.
+    """
+    # get patient ids in given cancer type 
+    gene, tissue = identifier.split('_')
+    tissue_ids = (sample_freeze_df
+        .query('DISEASE == @tissue')
+        .SAMPLE_BARCODE
+    )
+    
+    # get mutation and copy status
+    mutation_status = mutation_df.loc[tissue_ids, gene]
+    if cancer_classification == 'TSG':
+        copy_status = copy_loss_df.loc[tissue_ids, gene]
+    elif cancer_classification == 'Oncogene':
+        copy_status = copy_gain_df.loc[tissue_ids, gene]
+        
+    # get hit groups from mutation/CNV data
+    two_hit_samples = (mutation_status & copy_status).astype(int)
+    one_hit_samples = (mutation_status | copy_status).astype(int)
+        
+    return pd.DataFrame(
+        {'group': one_hit_samples + two_hit_samples}
+    )
+
+
+# In[15]:
+
+
+identifier = 'ATRX_LGG'
```

Comment:
Are you using this same notebook to rerun a bunch of cancer/gene pairs? Is there a way in python to perform these analyses and write them to a different output file passed in as a parameter? You can do a similar thing in RMarkdown.

Reply:
I am - I think this can be done using [jupytext](https://jupytext.readthedocs.io/en/latest/install.html), or maybe in other ways. Created #12 to think about this in the future.